sapl/protocoloadm/views.py

```diff
@@ -1607,3 +1611,43 @@ def filtra_tramitacao_destino_and_status(self, status, destino):
             status=status,
             unidade_tramitacao_destino=destino).distinct().values_list(
                 'documento_id', flat=True)
+
+
+def apaga_protocolos(request, ano):
+    all_protocolos = Protocolo.objects.filter(ano__in=ano)
+
+    for doc in DocumentoAdministrativo.objects.filter(protocolo__in=all_protocolos):
+        doc.protocolo = None
+        doc.save()
+
+    for ml in MateriaLegislativa.objects.filter(ano__in=ano):
+        ml.numero_protocolo = None
+        ml.save()
+
+    for deleted_object in all_protocolos:
+        post_delete_signal.send(sender=None,
+                                instance=deleted_object,
+                                operation='D',
+                                request=request
+                             )
+
+    all_protocolos.delete()
```

Comment:
Tem que ser o inverso: primeiro vc deleta os protocolos daí vc manda os sinais de deleção pra cada um deles.

Reply:
Se for colocado primeiro que os sinais a variável `all_protocolos` vira uma queryset vazia, o que faz com que não entre no `for` do sinal.